### ISSUE TITLE
[Bug]: Conversion is failed for: aten::max_pool2d, prim::ListConstruct

### ISSUE THREAD

**Issue by jikechao** [openvinotoolkit/openvino#20668]
### OpenVINO Version

2023.1.0-12185-9e6b00e51cd-releases/2023/1

### Operating System

Ubuntu 18.04 & Windows System

### Device used for inference

CPU

### Framework

PyTorch

### Issue description

Conversion crashed when I converted the following Torch model into OpenVINO using `ov.convert_model` API.

### Step-by-step reproduction
The simplest code for reproducing this crash is as follow:
```
import torch
from torch.nn import Module
import openvino as ov


class max_pool2d(Module):
    def forward(self, *args):
        return torch.nn.functional.max_pool2d(args[0], kernel_size=3)

torch_model = max_pool2d().float().eval()
input_data = torch.randn([7, 176, 3, 10], dtype=torch.float32)
trace = torch.jit.trace(torch_model, [input.clone() for input in input_data])
input_shapes = list([inp.shape for inp in input_data])
# print(input_shapes)

ov_model = ov.convert_model(trace, input=input_shapes)
```

### Relevant log output

```shell
Check 'is_conversion_successful' failed at src\frontends\pytorch\src\frontend.cpp:138:
FrontEnd API failed with OpConversionFailure:
Model wasn't fully converted. Failed operations detailed log:
-- aten::avg_pool3d with a message:
Exception happened during conversion of operation aten::avg_pool3d with schema aten::avg_pool3d(Tensor self, int[3] kernel_size, int[3] stride=[], int[3] padding=0, b
ool ceil_mode=False, bool count_include_pad=True, int? divisor_override=None) -> Tensor
Check 'input' failed at src\frontends\pytorch\srcode_context.cpp:156:
FrontEnd API failed with GeneralFailure:
Input with index 2 cannot be interpreted as Constant: util::PtFrameworkNode prim::ListConstruct_1 () -> (dynamic[...])

-- prim::ListConstruct with a message:
Exception happened during conversion of operation prim::ListConstruct with schema (no schema)
Check 'get_input_size() >= 1' failed at src\core\src\op\concat.cpp:34:
While validating node 'opset1::Concat Concat_13 () -> (dynamic[...])' with friendly_name 'Concat_13':
At least one argument required.

Summary:
-- Conversion is failed for: aten::avg_pool3d, prim::ListConstruct
```

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by vurusovs** [comment]
@rkazants @gkrivor please, take a look

**Comment by rkazants** [comment]
@mvafin, @eaidova, please take a look

**Comment by jikechao** [comment]
All pooling-related operators have the same crash, including `avg_pool2d`, `avg_pool3d`, `max_pool2d`, `max_pool3d`, `adaptive_avg_pool2d` and so on.

**Comment by mvafin** [comment]
@jikechao There are 2 issues I can see in this example. First because you were tracing before calling `convert_model` we didn't call `torch.jit.freeze` for you model, and because of that list input to pool is not converted to constant.
Second problem is that you provide input to model as list, not as tensor `[input.clone() for input in input_data] and use first element as input to max_pool2d and that is a tensor of shape `[176, 3, 10]` and we currently assume that input shape to pool2d operations is 4-dimensional, except for adaptive_pool operations, where we recently fixed that issue in #20350. So question is do you require such use case for your model?

**Comment by jikechao** [comment]
@mvafin Thanks for your kind response!
For the first issue:
The crash message disappeared after the statement `trace = torch.jit.freeze(trace)` was used.
What confuses me is that when I modify the above script and use any other operators (such as `softplus`), the script without `torch.jit.freeze` can convert to OV IR correctly. 
I want to know why only the model with the pooling operator must be frozen before conversion.

**Comment by mvafin** [comment]
- Problem that you have with `UNSPECIFIED` format: that is usually reported when input type was not deduced by conversion. You need to specify type in `input` argument. That will not happen when you provide `example_input` because in that case `convert_model` will automatically use types from example.
- What confuses me is that when I modify the above script and use any other operators (such as softplus), the script without torch.jit.freeze can convert to OV IR correctly. - `torch.jit.freeze` does some optimizations of the graph that sometimes helps us to convert models. In pool case it is a `list` is provided to pads/strides/dilations. softmax do not accept lists so no such issue.

**Comment by avitial** [comment]
Closing this, I hope previous responses were sufficient to help you proceed. Feel free to reopen and ask additional questions related to this topic.
